src/main/java/com/purbon/kafka/topology/AccessControlManager.java

```diff
@@ -30,30 +32,23 @@
 
   private AccessControlProvider controlProvider;
   private ClusterState clusterState;
-  private Map<String, String> cliParams;
+  private Configuration config;
 
   public AccessControlManager(AccessControlProvider controlProvider) {
-    this(controlProvider, new ClusterState(), new HashMap<>());
-  }
-
-  public AccessControlManager(
-      AccessControlProvider controlProvider, Map<String, String> cliParams) {
-    this(controlProvider, new ClusterState(), cliParams);
+    this(controlProvider, new ClusterState());
   }
 
   public AccessControlManager(AccessControlProvider controlProvider, ClusterState clusterState) {
-    this(controlProvider, clusterState, new HashMap<>());
+    this(controlProvider, clusterState, new MapConfiguration(new HashMap<>()));
   }
 
   public AccessControlManager(
-      AccessControlProvider controlProvider,
-      ClusterState clusterState,
-      Map<String, String> cliParams) {
+      AccessControlProvider controlProvider, ClusterState clusterState, Configuration config) {
     this.controlProvider = controlProvider;
     this.clusterState = clusterState;
-    this.cliParams = cliParams;
+    this.config = config;
 
-    this.allowDelete = Boolean.valueOf(cliParams.getOrDefault(ALLOW_DELETE_OPTION, "true"));
+    this.allowDelete = config.getBoolean(ALLOW_DELETE_OPTION, true);
```

Comment:
I think we can take the default value out of here as the default value will come anyhow from the CLI class.

Reply:
From what I found, no default value actually appears to break a few unit tests. So would potentially require some refactor against your tests to remove it.